api/python/cellxgene_census/tests/experimental/ml/test_pytorch.py

```diff
@@ -570,6 +570,33 @@ def test__shuffle(soma_experiment: Experiment) -> None:
     assert X_values == soma_joinids
 
 
+# noinspection PyTestParametrized,DuplicatedCode
+@pytest.mark.parametrize("obs_range,var_range,X_value_gen", [(16, 1, pytorch_seq_x_value_gen)])
+def test__shuffle_chunks(soma_experiment: Experiment) -> None:
+    dp = ExperimentDataPipe(
+        soma_experiment,
+        measurement_name="RNA",
+        X_name="raw",
+        obs_column_names=["label"],
+        shuffle=True,
+        soma_chunk_size=4,
+        shuffle_chunk_count=2,
+    )
+
+    all_rows = list(iter(dp))
+
+    soma_joinids = [row[1][0].item() for row in all_rows]
+    X_values = [row[0][0].item() for row in all_rows]
+
+    # same elements
+    assert set(soma_joinids) == set(range(16))
+    # not ordered! (...with a `1/16!` probability of being ordered)
+    assert soma_joinids != list(range(16))
```

Comment:
TODO: Could assert that the first and second half of the `soma_joinids` are each formed from exactly 2 quarters of the full data set (without knowing which ones specifically, since it's random).